downloads/swan-lake-release-notes/swan-lake-beta4/RELEASE_NOTE.md

```diff
@@ -0,0 +1,827 @@
+    }
+    
+    class Employee {
+        *Person; // Will now result in an error.
+        int id;
+
+        function init(string firstName, string lastName, string dob, int id) {
+            self.firstName = firstName;
+            self.lastName = lastName;
+            self.dob = dob;       
+            self.id = id;
+        }
+
+        private function getName() returns string => self.firstName;	
+    }
+    ```
+
+- Fixed a bug that resulted in compilation errors not being emitted for invalid `xml` template expressions.
+
+    ```ballerina
+    xml x = xml `</>`; // Will now result in an error.
+    ```
+
+- Fixed a bug that resulted in compilation errors not being emitted for duplicate fields written with escape sequences in the mapping constructor.
+
+    ```ballerina
+    map<any> x = {a\\: 454, "a\\": false}; // Will now result in an error.
+    ```
+
+- Updated the `xml:createElement` to accept the attribute map as the second argument.
+
+    ```ballerina
+    xml:Element jo = xml:createElement("name", {id: "1234"}, xml `Jo`);
+    ```
+
+- Updated the `xml:get` function’s return type to return the exact type `T` when the `xml` sequence is of type `xml<T>`.
+
+    ```ballerina
+    xml<xml:Element> employees = xml `<e1><name>Jo</name></e1><e2><name>Mary</name></e2>`;
+    xml:Element employee1 = employees.get(0);// Now allowed.
+    ```
+
+- Updated the `table:map` function’s function argument `func` and the return type to work with subtypes of mapping types instead of any type.
+
+    ```ballerina
+    table<record {int id; string name;}> tb = table [
+    {id: 1234, name: "Jo"},
+    {id: 2345, name: "May"}
+    ];
+
+    var idTable = tb.map(function (record {int id; string name;} r) returns int { // No longer allowed.
+    return r.id;
+    });
+    ```
+
+- Corrected a few deviations in the `lang.error` module according to the language specification. The `CallStack` class and `CallStackElement` records have been removed. Now, a stack frame is represented by an `error:StackFrame` object.
+	
+    Prior to Swan Lake Beta4, the `error:stackTrace()` function returned an `error:CallStack` object which had the structure below which was a deviation from the specification.
+
+    ```ballerina
+    public class CallStack {
+    public CallStackElement[] callStack = [];
+    }
+    ```
+
+    This has now been fixed, and it is no longer possible to retrieve an `error:CallStack` object or directly access the `callStack` array as shown below.
+
+    ```ballerina
+    error:CallStack callStack = err.stackTrace(); // `CallStack` is undefined
+    error:CallStackElement[] elements = err.stackTrace().callStack; // Not allowed, `CallStackElement` is undefined, no `callStack` field.
+    ```
+
+    The `error:stackTrace` function now returns an array of `StackFrame` objects.
+
+    ```ballerina
+    public type StackFrame readonly & object {
+    public function toString() returns string;
+    };
```

Comment:
This code block is not properly formatted

Reply:
Fixed.